DiscordBot.DataAccess/EventsSheetsService.cs

```diff
@@ -576,5 +629,15 @@ IEnumerable<EventResponse> responseColumns
 
             await request.ExecuteAsync();
         }
+
+        private IEnumerable<EventResponse> ParseResponseHeaders(IList<IList<object>> sheetsResponse)
```

Comment:
This file is getting very long again. Might be worth extracting all the parsing to a parser class? 🙂

Reply:
Yeah the class is quite large. I've extracted the parsing methods as you've suggested in 73aa63b, but it's still quite large. Nearly all of the remaining methods use some of the `EventsSheetsService` state, usually the underlying Google Sheets service class. 

I could try to group these methods together into classes, but they would either need to take most of this `EventsSheetsService`'s fields as parameters, or contain the `EventSheetService` state inside their own classes. Is this a good idea? Or should I just stick with this large class file 🙂

Reply:
Hm, good question. I think extracting the parsing and request creation is a good start because those are completely separate concerns that it's good to split out. So just make sure that any logic to do with creating requests or parsing responses is moved out to those classes. Otherwise I think it's probably fine to keep the rest in this class at least for now. 

Just keep your eye out for anything else which can be split out as you're working. I don't think needing to pass the underlying service class or any other state as parameters is necessarily the end of the world, but it's important that the new class is genuinely separate to this service class. So that it has a separate purpose and isn't just an extension of this class, if that makes sense 🙂

Reply:
Yeah, will do! Thanks for that 🙂